src/Core/Worker/Repository/Process.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2021, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Core\Worker\Repository;
+
+use Friendica\BaseRepository;
+use Friendica\Core\Worker\Exception\ProcessPersistenceException;
+use Friendica\Database\Database;
+use Friendica\Util\DateTimeFormat;
+use Friendica\Core\Worker\Factory;
+use Friendica\Core\Worker\Entity;
+use Psr\Log\LoggerInterface;
+
+/**
+ * functions for interacting with a process
+ */
+class Process extends BaseRepository
+{
+	protected static $table_name = 'process';
+
+	/** @var Factory\Process */
+	protected $factory;
+
+	public function __construct(Database $database, LoggerInterface $logger, Factory\Process $factory)
+	{
+		parent::__construct($database, $logger, $factory);
+	}
+
+	/**
+	 * Starts and Returns the process for a given PID
+	 *
+	 * @param int $pid
+	 *
+	 * @return Entity\Process
+	 */
+	public function create(int $pid): Entity\Process
+	{
+		// Cleanup inactive process
+		$this->deleteInactive();
+
+		try {
+			$this->db->transaction();
+
+			$trace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS);
+			$last  = $trace[count($trace) - 1];
+
+			$command = strtolower(basename($last['file']));
```

Comment:
Shouldn't this be a method parameter where we can pass `basename(__FILE__)` in the call instead?